python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -0,0 +1,1023 @@
+            pl_expr.StringFunction.Uppercase,
+            pl_expr.StringFunction.EndsWith,
+            pl_expr.StringFunction.StartsWith,
+        ):
+            raise NotImplementedError(f"String function {self.name}")
+
+    @with_mapping
+    def evaluate(
+        self,
+        df: DataFrame,
+        *,
+        context: ExecutionContext = ExecutionContext.FRAME,
+        mapping: dict[Expr, Column] | None = None,
+    ) -> Column:
+        """Evaluate this expression given a dataframe for context."""
+        columns = [
+            child.evaluate(df, context=context, mapping=mapping)
+            for child in self.children
+        ]
+        if self.name == pl_expr.StringFunction.Lowercase:
+            (column,) = columns
+            return Column(plc.strings.case.to_lower(column.obj), column.name)
+        elif self.name == pl_expr.StringFunction.Uppercase:
+            (column,) = columns
+            return Column(plc.strings.case.to_upper(column.obj), column.name)
+        elif self.name == pl_expr.StringFunction.EndsWith:
+            column, suffix = columns
+            return Column(
+                plc.strings.find.ends_with(column.obj, suffix.obj), column.name
+            )
+        elif self.name == pl_expr.StringFunction.StartsWith:
+            column, suffix = columns
+            return Column(
+                plc.strings.find.starts_with(column.obj, suffix.obj), column.name
+            )
+        else:
+            raise NotImplementedError(f"StringFunction {self.name}")
+
+
+class Sort(Expr):
+    __slots__ = ("options", "children")
+    _non_child = ("dtype", "options")
+
+    def __init__(
+        self, dtype: plc.DataType, options: tuple[bool, bool, bool], column: Expr
+    ):
+        super().__init__(dtype)
+        self.options = options
+        self.children = (column,)
+
+    @with_mapping
+    def evaluate(
+        self,
+        df: DataFrame,
+        *,
+        context: ExecutionContext = ExecutionContext.FRAME,
+        mapping: dict[Expr, Column] | None = None,
+    ) -> Column:
+        """Evaluate this expression given a dataframe for context."""
+        (child,) = self.children
+        column = child.evaluate(df, context=context, mapping=mapping)
+        (stable, nulls_last, descending) = self.options
+        order, null_order = sorting.sort_order(
+            [descending], nulls_last=nulls_last, num_keys=1
+        )
+        do_sort = plc.sorting.stable_sort if stable else plc.sorting.sort
+        table = do_sort(plc.Table([column.obj]), order, null_order)
+        return Column(table.columns()[0], column.name).set_sorted(
+            is_sorted=plc.types.Sorted.YES, order=order[0], null_order=null_order[0]
+        )
+
+    def collect_agg(self, *, depth: int) -> AggInfo:
+        """Collect information about aggregations in groupbys."""
+        # TODO: Could do with sort-based groupby and segmented sort post-hoc
+        raise NotImplementedError("Sort in groupby")
```

Comment:
OK yeah based on what I'm seeing with these I think you want to use my suggestion above for printing more descriptive NotImplementedErrors straight from the parent class.